packages/jsActions/mobile-resources-native/src/file-download/DownloadFile.ts

```diff
@@ -0,0 +1,86 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import { Platform } from "react-native";
+import RNBlobUtil from "react-native-blob-util";
+import FileViewer from "react-native-file-viewer";
+import mimeTypes from "mime";
+
+// BEGIN EXTRA CODE
+function formatPath(...pathArgs: string[]): string {
+    return pathArgs.filter(arg => !!arg).join("/");
+}
+
+function sanitizeFileName(name: string) {
+    /* eslint-disable-next-line no-control-regex */
+    return name.replace(/[<>"?:|*/\\\u0000-\u001F\u007F]/g, "_");
+}
+
+function getFilePathWithNextVersion(path: string) {
+    const fileVersion = path.match(/\(\d+\)(?!.*\(\d+\))/);
+    if (!fileVersion) {
+        return path.replace(/\.(?!.*\.)/, "(1).");
+    }
+    const newFileVersion = `(${Number(fileVersion[0].match(/\d+/)?.[0]) + 1})`;
+    return path.replace(/\(\d+\)(?!.*\(\d+\))/, newFileVersion);
+}
+
+async function getUniqueFileName(path: string) {
+    let uniqueFilePath = path;
+    do {
+        uniqueFilePath = getFilePathWithNextVersion(uniqueFilePath);
+    } while (await RNBlobUtil.fs.exists(uniqueFilePath));
+    return uniqueFilePath;
+}
```

Comment:
This seems unnecessarily complex. Can't we just take the original path and insert a `(1)`, `(2)` etc. into it with a simple counter, instead of parsing each previous attempt? Also, I would expect a `while` instead of a `do ... while` here, as we don't need to do anything if the file does not exist the first time around.

Reply:
The new version still looks very complicated, with a lot of difficult-to-read regexes. We only need to determine the insertion index (= at the last dot after the last path separator) once, after which we could simply do something like this:

```
    const pathWithoutExtension = path.substring(0, insertionIndex);
    const extension = path.substring(insertionIndex);

    let uniqueFilePath = path;
    let version = 0;

    while (!(await RNBlobUtil.fs.exists(uniqueFilePath))) {
        uniqueFilePath = `${pathWithoutExtension } (${++version})${extension}`;
    }

    return uniqueFilePath;
```

Reply:
If we pass the directory and the file name to this method separately and do the `formatPath` here, we don't have to worry about path separators when determining the insertion index. I think that makes more sense than first combining the path and _then_ messing with it to come up with a unique file name.

Reply:
It looks much simpler now :) ✨